daml-script/dump/src/main/scala/com/daml/script/dump/Encode.scala

```diff
@@ -254,14 +280,30 @@ private[dump] object Encode {
       tree: TransactionTree,
   ): Doc = {
     val rootEvs = tree.rootEventIds.map(tree.eventsById(_).kind)
-    val submitters = rootEvs.flatMap(evParties(_)).toSet
-    val cids = treeCreatedCids(tree)
-    val treeBind =
-      (Doc.text("tree <- submitTreeMulti ") + encodeParties(partyMap, submitters) + Doc.text(
-        " [] do"
-      ) / Doc.stack(rootEvs.map(ev => encodeEv(partyMap, cidMap, ev)))).hang(2)
-    val cidBinds = cids.filter(c => cidRefs.contains(c.cid)).map(bindCid(cidMap, _))
-    Doc.stack(treeBind +: cidBinds)
+    val createsOnly = {
+      import scalaz.Scalaz._
+      rootEvs.toList
+        .map {
```

Comment:
`.map_.sequence` should be equivalent to `.traverse`, maybe also worth factoring into a helper function.